src/dune_pkg/local_package.ml

```diff
@@ -108,16 +64,82 @@ module For_solver = struct
          (Package_dependency.list_to_opam_filtered_formula depopts)
   ;;
 
-  let opam_filtered_dependency_formula { dependencies; _ } =
-    Package_dependency.list_to_opam_filtered_formula dependencies
+  let opam_filtered_dependency_formula { dependencies; _ } = dependencies
+
+  module T = struct
+    type t = Package_name.t * OpamTypes.condition
+
+    let to_dyn = Dyn.pair Package_name.to_dyn Opam_dyn.condition
+
+    let compare (name, condition) (other_name, other_condition) =
+      let open Ordering.O in
+      let= () = Package_name.compare name other_name in
+      let compare_filters filter filter' =
+        let get_vars = function
+          | OpamTypes.Constraint _ -> []
+          | Filter filter -> Stdlib.List.sort Stdlib.compare (OpamFilter.variables filter)
+        in
+        match get_vars filter, get_vars filter' with
+        | v :: _, v' :: _ -> Stdlib.compare v v'
+        | [], _ :: _ -> 1
```

Comment:
We don't use numbers for orderings in the dune code, just the constructors of `Ordering`.

Reply:
That's true and I personally also prefer this, but this function gets passed to `OpamFormula.compare_formula` which requires the integer way of comparing. I can reimplement `compare_formula` to use `Ordering` or add a new function to `Ordering` that would convert comparison functions between the representations. Which of those do you think would be the best approach?

Reply:
That code just went away completely, fortunately.